prow/tide/tide.go

```diff
@@ -300,8 +303,53 @@ type manager interface {
 	GetFieldIndexer() ctrlruntimeclient.FieldIndexer
 }
 
+type Controller struct {
+	syncCtrl   *syncController
+	statusCtrl *statusController
+}
+
+// Shutdown signals the statusController to stop working and waits for it to
+// finish its last update loop before terminating.
+// Controller.Sync() should not be used after this function is called.
+func (c *Controller) Shutdown() {
+	c.syncCtrl.Shutdown()
+	c.statusCtrl.Shutdown()
+}
+
+func (c *syncController) Shutdown() {
+	c.History.Flush()
+}
+
+func (c *statusController) Shutdown() {
+	c.shutdown()
+}
```

Comment:
nit: These functions don't seem to add anything, they just make this more confusing to read.

Reply:
deleted